### PR TITLE
add dynamic contact attributes to brevo contact

### PR DESCRIPTION
**Pull request by RainbowBunchie**
This adds dynamic contact attributes to the module configuration to mirror the attribute configurations in brevo and add the correct types to the module. 

This is part of COM-267

### CODE REVIEW

File: demo/api/src/brevo-contact/dto/brevo-contact-attributes.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    @IsNotEmpty()
+    @IsString()
+    @Field()
+    LASTNAME: string;
+
+    @IsUndefinable()
+    @IsNotEmpty()
+    @IsString()
+    @Field()
+    FIRSTNAME: string;
```
Comment (raphaelblum) [inline demo/api/src/brevo-contact/dto/brevo-contact-attributes.ts:18]: here as well

File: packages/api/src/config/brevo-module.config.ts [hunk 2]
```diff
@@ -1,5 +1,8 @@
+import { BrevoContactAttributesInterface } from "src/types";
+
 export interface BrevoModuleConfig {
     brevo: {
         apiKey: string;
+        BrevoContactAttributes: BrevoContactAttributesInterface;
```
Comment (raphaelblum) [inline packages/api/src/config/brevo-module.config.ts:6]: should the attributes be required?
in the packages/api/src/brevo-api/brevo-api-contact.service.ts and packages/api/src/brevo-contact/dto/brevo-contact.factory.ts the attributes are optional.

I think they could be optional, but if not I'd say they should be required in the 2 files mentioned above.

File: demo/api/src/brevo-contact/dto/brevo-contact-attributes.ts (hunk 1 above)
Comment (RainbowBunchie) [reply demo/api/src/brevo-contact/dto/brevo-contact-attributes.ts:16]: I removed the decorator in [20ec2d4](https://github.com/vivid-planet/comet-brevo-module/pull/6/commits/20ec2d4bbc50ade332021d9b17b38be40b412a35)

File: demo/api/schema.gql [hunk 3]
```diff
@@ -339,11 +326,33 @@ type PaginatedDamFolders {
...
 }
 
+type BrevoContact {
+  id: Int!
+  createdAt: String!
+  modifiedAt: String!
+  email: String!
+  emailBlacklisted: Boolean!
+  smsBlacklisted: Boolean!
+  listIds: [Float!]!
```
[the remark this answers is not in the document]
Comment (RainbowBunchie) [reply demo/api/schema.gql:336]: Yes, i will create a separate PR right now for this.

File: demo/api/schema.gql [hunk 4]
```diff
@@ -339,11 +326,33 @@ type PaginatedDamFolders {
...
+  listUnsubscribed: [Float!]!
```
[the remark this answers is not in the document]
Comment (RainbowBunchie) [reply demo/api/schema.gql:337]: please check #9

File: packages/api/src/brevo-contact/dto/brevo-contact.factory.ts [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+
+            @Field(() => [Number])
+            listIds: number[];
+
+            @Field(() => [Number])
+            listUnsubscribed?: number[];
+        }
+
+        if (BrevoContactAttributes) {
+            @ObjectType({})
```
Comment (raphaelblum) [inline packages/api/src/brevo-contact/dto/brevo-contact.factory.ts:48]:
```suggestion
            @ObjectType()
```

**Review by raphaelblum — APPROVED** [review]

**Review by raphaelblum — APPROVED** [review]

**Pull request merged**